packages/cspell-types/src/CSpellReporter.ts

```diff
@@ -19,29 +19,15 @@ export const MessageTypes: MessageTypeLookup = {
     Warning: 'Warning',
 };
 
-export interface MessageEmitter {
-    (message: string, msgType: MessageType): void;
-}
-
-export interface DebugEmitter {
-    (message: string): void;
-}
+export type MessageEmitter = (message: string, msgType: MessageType) => void;
 
-type ErrorLike = Error | { message: string; name: string; toString: () => string };
+export type DebugEmitter = (message: string) => void;
```

Comment:
I think you are right with the `Promise<void>` return. Everything becomes a bit simpler.